plugins/module_utils/errors.py

```diff
@@ -21,3 +21,7 @@ class SyncError(Error):
 
 class SensuError(Error):
     """ Error that signals problems with Sensu Go web API. """
+
+
+class BonsaiError(Error):
+    """ Error that signals problems with Bonsai assets. """
```

Comment:
The general idea behind the exception naming scheme was to make it easy to identify what part is responsible for the failure. For example, `BonsaiError` indicates that something with external Bonsai service went haywire and `SensuError` should point to interactions with Sensu Go backend. What I do find out of place is that `SyncError`, since this clearly belongs to the `SensuError` category.

Maybe we should structure them into only three groups:

 1. `HttpError` indicates connection problems (cannot connect and other low-level stuff).
 2. `SensuError` indicates logical problems with Sensu API.
 3. `BonsaiError` indicates logical problems with Bonsai API.

This, combined with a custom `__str__` method on the `Error` class, should give us a much better idea where the error stems from.

That being said, maybe we can fix this in a follow-up PR? This change will not affect the public API in any way, so this should be safe to postpone a bit.

Reply:
Hmm, indeed, if I see `SyncError`, then I would also expect `AuthError` and the like. And if this were the case, we would quickly find ourselves in a situation where we have different error type for alsmost every error. Grouping errors with respect to domains (Bonsai, Sensu) and removing error types that are off rather than adding new ones, does make sense.

Yes, let's keep it as it is for now - regardless of the error types that we're currently using, the error messages themselves are descriptive enough. We can come back to it at a later time and revisit the naming scheme.